service/process.go

```diff
@@ -191,6 +191,12 @@ func (this *service) processIncoming(msg message.Message) error {
 func (this *service) processAcked(ackq *sessions.Ackqueue) {
 	for _, ackmsg := range ackq.Acked() {
 		// Let's get the messages from the saved message byte slices.
+
+		if len(ackmsg.Msgbuf) == 0 || len(ackmsg.Ackbuf) == 0 {
+			glog.Errorf("process/processAcked: Unable to decode new %s as ackmsg Msgbuf buf is %v or ackmsg Ackbuf buf is %v", ackmsg.Mtype, ackmsg.Msgbuf, ackmsg.Ackbuf)
```

Comment:
If it's not an issue, I'll probably just make it:
```
if (len(ackmsg.Msgbuf) == 0 || len(ackmsg.Ackbuf) == 0) && ackmsg.Mtype!="PINGREQ"
```

But so far I have not had the i/o timeout issues I was having before, so I really like this (thanks for this PR!)